### ISSUE TITLE
Skip services in the home visit form when child is registered above a certain age

### ISSUE THREAD

**Issue by cafootitt** [OpenSRP/opensrp-client-chw#143]
For Liberia, most of the vaccines expire at 11 months of age (the exception is OPV 0 that expires at 7 days and MCV 2, which expires at 23 months).

I'm thinking that at some age cutoff when children are registered, we wouldn't even want to show any immunizations anymore. Something like 24 months of age perhaps(?), because if a child is registered when they're already 2 years of age, they are too old to receive vaccines anyway. This cutoff age would vary between countries though, so this is something we'd want to be configurable. For e.g. in Zambia, they give vaccines up to age 5.

To pre-empt the question of why we can't just always show the vaccines that are due for that child's age. For e.g. a 10 week old is registered, why not just skip birth and 6 weeks and show 10 week vaccines. The reason for this is because of the constraints, i.e. 10 week vaccines can't become due until 6 week vaccines are recorded. I don't know if we want to get into more complexity here of overriding this constraint in this circumstance, but I was thinking we could start with the suggestion above as a first improvement.

Thoughts?

I guess the same goes for MNP (all expire at 24 months of age), Vitamin A is NA because it expires at 60 months of age, and same for deworming (60 months)

**Comment by rowo** [comment]
@cagulas What do you expect to happen in that case where a 10 week old gets registered? Would they backfill all of the vaccines back to birth and then follow up up? And you are thinking for 2 year olds, it's not important to backfill that data because there's no future vaccines to plan for and the data isn't shared with anyone else?

**Comment by cagulas** [comment]
@rowo Yes, that's my thinking exactly.

**Comment by rowo** [comment]
I'm all for not making CHWs fill in unnecessary data. 

For children registered under 2, does the vaccine section get skipped once all vaccines are complete or only when expired on the home visit form?

Also unrelated and I should take a look, but is the backfilling data process easy?

**Comment by cagulas** [comment]
@rowo The backfilling data process is confusing right now: https://github.com/OpenSRP/opensrp-unicef-wcaro/issues/142. One thing I'd like to discuss today.

If I understood your question correctly, yes, when they backfill immunizations and record all vaccinations and the child is up to date, the vaccines don't appear in the home visit anymore.

**Comment by cagulas** [comment]
@keymane Roger and I discussed this yesterday, and we'd like to propose that we add a feature, whereby if the service schedule has already expired for the child, based on their age at the time they're registered, then we wouldn't show those services in the home visit.

For example:
1) If a child is registered in the app when they're 11 months old (11 months is when all of the first year vaccines expire, then we wouldn't show any due vaccines, as they're already too old to receive any of them. 

**Side note on this reasoning:** One could argue that it would be worth recording their vaccine history, so we have that data, but this isn't a use case we need to support. Our use case is to support timely service delivery for all future services the child is eligible to receive. Furthermore, when we calculate the fully immunized children indicator, we wouldn't want to artificially deflate or inflate the indicator using vaccine data from before the app is deployed, as that won't be a true reflection of the impact of the app on getting children immunized.

2) At 23 months of age, MCV 2 expires, so if a child is registered at 23 months of age or above, we wouldn't show MCV 2.

3) OPV 0 is somewhat of an exception. It expires at 7 days of age, although I hesitate to say we should not show it in the app if the child is registered when they're 7 days old, because the registration of the child might not always happen within the first week, especially as the registration would likely only happen during the next home visit from the CHW worker. If we did this, we might never get any OPV 0's recorded when in fact they are getting them at the health facility after delivery. 

I propose for OPV 0, we show it along with BCG in birth immunizations for the first home visit after a child is registered. If it's marked as not given, then I propose we let it expire and it wouldn't show up again (as long as the child is >= 7 days of age).

4) Exclusive breastfeeding. This should already be implemented right now like this, but just to reiterate that these expire at the end of each month in months 0-5. For e.g. if a child is registered at birth, we show exclusive breastfeeding 0 months. If child is registered at 1 month of age, we show exclusive breastfeeding at 1 month of age and so on. If a child is registered at 6 months of age, they don't see exclusive breastfeeding at all.

5) MNP - If a child is registered at 24 months of age, don't show MNP anymore.

6) For Vita A and deworming, these all expire at 60 months, so we don't have to do anything special here, because 60 months is when the child leaves the child register.

7) Birth certificate doesn't expire, so NA

8) Obs & Illness episodes is optional always, so NA

@keymane @rowo Pls review the above and let me know your thoughts.

**Comment by cagulas** [comment]
Updated the issue above to mention OPV 0 expiry will be at one month. We discussed on a call today with the team that it will be difficult to check each time for OPV 0 whether it's the first visit or not, so just putting an expiration age for OPV 0 is the best approach. Since 7 days is too short, we'll put it at one month.

**Comment by rowo** [comment]
@cagulas Yes, I think this covers what we discussed at a high level. 

@keymane Rather than general thoughts, I think we are looking specifically for your general assessment of effort to build this feature, "hide services in the home visit form if the service has expired for the child based on their age at registration".

**Comment by cagulas** [comment]
@keymane I think the expiry dates are included as part of issue 189, or do we need to create a new issue saying that children above a certain age should not be enrolled in these service schedules? I'm hoping that's already covered though by the expires in 189. Please let me know.

**Comment by keymane** [comment]
@cagulas @rowo This would be hard to implement. We still use motech schedules for vaccines and recurring services and the prerequisite is the date of birth or a previous vaccine. If the previous vaccine the susequent  vaccines cannot be due. 

If we convert the vaccines and schedules to rules engine (https://github.com/OpenSRP/opensrp-client-child/issues/12), it may be easier to implement.

**Comment by cagulas** [comment]
@keymane What about adding a condition that the child is less than the expiry date for the schedule to be created, similar to how we saw conditions for OPV 4 and MR in zeir motech schedules?

**Comment by keymane** [comment]
@cagulas The issue is that some vaccines have previous vaccines as prerequisites. So if the previous vaccine is not given because it expired then the next one cannot be due. The conditions were a-bit complex and required testing alot. I would suggest we convert the schedules to use the rules engine and we can better options.

**Comment by cagulas** [comment]
After speaking to @keymane on this and explaining the expiries, the prerequisites is not an issue as it turns out, but the recommendation is still to wait to do this until after we convert the motech schedules to the new rules engine, as it will be easier.

**Comment by cagulas** [comment]
This is blocked by https://github.com/OpenSRP/opensrp-client-child/issues/12 as a result of the above discussion.

**Comment by cagulas** [comment]
@mahmud6390 When you're back from Geneva and the rules engine refactor is complete, you can work on this issue.

**Comment by cafootitt** [comment]
I found one issue with vaccines. I'll close this and create a new issue for that one. Good work!
